src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -213,16 +214,22 @@ export class DebugService implements IDebugService {
 		return this.initializing ? State.Initializing : State.Inactive;
 	}
 
-	private startInitializingState(): void {
+	get initializingOptions(): IDebugSessionOptions | undefined {
```

Comment:
This is the only part that is not straightforward. When the debug service first enters the 'initializing' state, we can't see the launch config that caused it to do that because it hasn't been resolved to a full debug session. We want to change the debug UI as soon as possible, so we have to expose the options from the new launch config somehow...

This is a bit odd but I'm not sure what a cleaner solution would be

Reply:
I'm a bit confused: the `simpleUI` flag is a property of `IDebugSessionOptions` and not part of a launch config. So why would a resolved launch config help with accessing the `simpleUI` flag?

But other than that your change looks reasonable.

Reply:
Right, once debugging has started I would get it from the DebugSession, but that doesn't exist yet, so at this stage it has to be exposed in a different way.

Reply:
Aha, you meant `DebugSession` and not "launch config", now I understand.